p2p/transport/webrtc/stream.go

```diff
@@ -150,15 +136,50 @@ func newStream(
 }
 
 func (s *stream) Close() error {
+	defer s.cleanup()
 	closeWriteErr := s.CloseWrite()
 	closeReadErr := s.CloseRead()
-	if closeWriteErr != nil {
-		return closeWriteErr
+	if closeWriteErr != nil || closeReadErr != nil {
+		s.Reset()
+		return errors.Join(closeWriteErr, closeReadErr)
 	}
-	return closeReadErr
+
+	s.mx.Lock()
+	s.controlMessageReaderEndTime = time.Now().Add(maxFINACKWait)
+	s.mx.Unlock()
+	s.SetReadDeadline(time.Now().Add(-1 * time.Hour))
+	<-s.controlMessageReaderDone
+	return nil
```

Comment:
Shouldn't this error if reading the ack fails ?
Like you can't actually fail to close a stream (unless maybe it's already closed) because even if the remote refuse to close you would still stop responding on this stream, so I always thought the `error` field was used to indicate already written data has been lost.